systemtests/generate_testdata_on_network.py

```diff
@@ -505,6 +505,136 @@ def do_run(args: Any):
             relayed_version=relayed_version,
         ), await_completion=True)
 
+    for relayed_version in relayed_v3_marker:
+        print(f"## Relayed v{relayed_version}, intra-shard MoveBalance, when relayer is same as sender")
+        controller.send(controller.create_relayed_with_move_balance(
+            relayer=accounts.get_user(shard=SOME_SHARD, index=0),
+            sender=accounts.get_user(shard=SOME_SHARD, index=0),
+            receiver=accounts.get_user(shard=SOME_SHARD, index=2).address,
+            amount=42,
+            additional_gas_limit=0,
+            relayed_version=relayed_version,
+        ), await_completion=True)
+
+    for relayed_version in relayed_v3_marker:
+        print(f"## Relayed v{relayed_version}, cross-shard MoveBalance, when relayer is same as sender")
+        controller.send(controller.create_relayed_with_move_balance(
+            relayer=accounts.get_user(shard=SOME_SHARD, index=0),
+            sender=accounts.get_user(shard=SOME_SHARD, index=0),
+            receiver=accounts.get_user(shard=OTHER_SHARD, index=0).address,
+            amount=42,
+            additional_gas_limit=0,
+            relayed_version=relayed_version,
+        ), await_completion=True)
+
+    for relayed_version in relayed_v3_marker:
+        print(f"## Relayed v{relayed_version}, intra-shard MoveBalance, with refund, when relayer is same as sender")
+        controller.send(controller.create_relayed_with_move_balance(
+            relayer=accounts.get_user(shard=SOME_SHARD, index=0),
+            sender=accounts.get_user(shard=SOME_SHARD, index=0),
+            receiver=accounts.get_user(shard=SOME_SHARD, index=2).address,
+            amount=42,
+            additional_gas_limit=42000,
+            relayed_version=relayed_version,
+        ), await_completion=True)
+
+    for relayed_version in relayed_v3_marker:
+        print(f"## Relayed v{relayed_version}, cross-shard MoveBalance, with refund, when relayer is same as sender")
```

Comment:
with refund, when relayer is same as sender

with refund, when relayer is same as sender and receiver